### ISSUE TITLE
Order of validators in chain (synchronous and asynchronous validators)

### ISSUE THREAD

**Issue by andrewsboev** [JeremySkinner/FluentValidation#1199]
### System Details

- FluentValidation version: 8.3.0

### Issue Description

Hi, it's kinda question rather than issue. 
According to method https://github.com/JeremySkinner/FluentValidation/blob/master/src/FluentValidation/Internal/PropertyRule.cs#L318, synchronous validators in chain are executed before asynchronous validators. 

For example, if I have such a chain: 

```
RuleFor(x => x.Amount).Cascade(CascadeMode.StopOnFirstFailure)
                      .NotNull()
                      .SetValidator(amountValidator)
                      .Must(x => x > 0);
```

And root validator for it is calling method `ValidateAsync` (it means that the context will be async, and `amountValidator` (which inherits `AbstractValidator<T>`) will be executed asynchronously).
The order of execution will be: NotNull(), Must(), and only then `amountValidator`. The purpose of `CascadeMode` is getting obscure, if order of validators is really undefined. 

It looks like a feature in code, not a bug. Why?

**Comment by JeremySkinner** [comment]
It's a holdover from how async support was initially implemented. It's something we can look at for a future version byt yes, you're correct - CascadeMode is really not useful when mixing sync & async validators in the same rule. It is much better to use conditions (`When`/`Unless`) to control which validators are executed anyway.

**Comment by andrewsboev** [comment]
Thanks for answer!

**Comment by JeremySkinner** [comment]
Fixed in 7a20491be719cc08d27f105a73cabc27960f50f1. This will be part of 8.5
